### PR TITLE
[FLINK-15198][Deployment / Mesos] remove mesos.resourcemanager.tasks.mem

### PR DESCRIPTION
**Pull request by PengTaoWW**
fix https://issues.apache.org/jira/projects/FLINK/issues/FLINK-15198

## What is the purpose of the change

In FLINK-15082, we deprecated 'mesos.resourcemanager.tasks.mem' in favour of the new unified option 'taskmanager.memory.total-process.size' from FLIP-49. We should remove it now in 1.11.
https://issues.apache.org/jira/projects/FLINK/issues/FLINK-15198?filter=allopenissues

## Brief change log

modified the code and deleted the part that used 'mesos.resourcemanager.tasks.mem' in the code

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
 	 */
 	@Deprecated
 	public static final ConfigOption<Integer> MESOS_RM_TASKS_MEMORY_MB =
-		key("mesos.resourcemanager.tasks.mem")
+		key("taskmanager.memory.total-process.size")
```
Comment (KarmaGYZ) [inline flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: I think we could not simply change the key.

Comment (PengTaoWW) [reply flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: from https://github.com/apache/flink/pull/10513/files#r356585015, just need to delete the code and comments that use MEMOS_RM_TASKS_MEMORY_MB?

**Comment by PengTaoWW** [comment]
@KarmaGYZ I modified the code and deleted the part that used 'mesos.resourcemanager.tasks.mem' in the code。

**Comment by PengTaoWW** [comment]
@flinkbot run travis

File: flink-jepsen/src/jepsen/flink/db.clj [hunk 2]
```diff
@@ -303,7 +303,7 @@
     "-Djobmanager.rpc.address=$(hostname -f)"
     "-Djobmanager.rpc.port=6123"
     "-Dmesos.resourcemanager.tasks.cpus=1"
-    "-Dmesos.resourcemanager.tasks.mem=2048" ;; FLINK-15082: this option must be set instead of taskmanager.memory.process.size
+    "-Dtaskmanager.memory.process.size=2048m" ;; FLINK-15082: this option must be set instead of taskmanager.memory.process.size
```
Comment (KarmaGYZ) [inline flink-jepsen/src/jepsen/flink/db.clj]: I guess this note could be removed after this PR.

File: flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java [hunk 3]
```diff
@@ -429,26 +418,15 @@ private static double getCpuCores(final Configuration configuration) {
 	}
 
 	private static MemorySize getTotalProcessMemory(final Configuration configuration) {
-		MemorySize legacyTotalProcessMemory = MemorySize.parse(configuration.getInteger(MESOS_RM_TASKS_MEMORY_MB) + "m");
 		MemorySize unifiedTotalProcessMemory = configuration.get(TaskManagerOptions.TOTAL_PROCESS_MEMORY);
 
-		if (configuration.contains(MESOS_RM_TASKS_MEMORY_MB) &&
-			configuration.contains(TaskManagerOptions.TOTAL_PROCESS_MEMORY) &&
-			!legacyTotalProcessMemory.equals(unifiedTotalProcessMemory)) {
+		if (!configuration.contains(TaskManagerOptions.TOTAL_PROCESS_MEMORY)) {
```
Comment (KarmaGYZ) [inline flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: I'm not sure that the total process memory should be explicitly set here. According to the [FLIP-49 Doc](https://cwiki.apache.org/confluence/display/FLINK/FLIP-49%3A+Unified+Memory+Configuration+for+TaskExecutors). User need either of these three options configured:
- Task Heap Memory and Managed Memory
- Total Flink Memory
- Total Process Memory

After this PR, I think we could directly construct the TaskExecutorResourceSpec from the flinkConfig. @azagrebin Please correct me if I have any misunderstanding.

File: flink-jepsen/src/jepsen/flink/db.clj (hunk 2 above)
Comment (PengTaoWW) [reply flink-jepsen/src/jepsen/flink/db.clj]: Let me fix it

**Comment by PengTaoWW** [comment]
@flinkbot run travis

**Comment by PengTaoWW** [comment]
@flinkbot run travis

**Comment by azagrebin** [comment]
@PengTaoWW 
We usually use `git rebase` on master instead of `merge` to fetch changes from `master` into PR.
I have done this for now, squashed the commits and force-pushed it into your branch.
Just for note, you will have to remove your branch locally and checkout it again from the updated remote one to work on this further.

File: flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java (hunk 3 above)
Comment (azagrebin) [reply flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: Indeed, we do not need this method `getTotalProcessMemory` anymore.
`TaskExecutorProcessUtils` will parse the options in `createContaineredTaskManagerParameters`.
Subsequently, we do not need `.withTotalProcessMemory(totalProcessMemory)` in `createContaineredTaskManagerParameters`.

Comment (PengTaoWW) [reply flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: I will remove .withTotalProcessMemory (totalProcessMemory) in createContaineredTaskManagerParameters and delete the getTotalProcessMemory method

**Comment by PengTaoWW** [comment]
@flinkbot run travis

**Comment by PengTaoWW** [comment]
@flinkbot run travis

File: flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java (hunk 3 above)
Comment (PengTaoWW) [reply flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java]: @azagrebin Already modified

File: docs/ops/memory/mem_migration.md [hunk 4]
```diff
@@ -134,7 +134,7 @@ but they still have the same semantics for the job manager process. See also [ho
 The previous options which were responsible for the total memory used by Flink are `taskmanager.heap.size` or `taskmanager.heap.mb`.
 Despite their naming, they included not only JVM heap but also other off-heap memory components. The options have been deprecated.
 
-The Mesos integration also had a separate option with the same semantics: `mesos.resourcemanager.tasks.mem` which has also been deprecated.
```
[the remark this answers is not in the document]
Comment (azagrebin) [reply docs/ops/memory/mem_migration.md]: As it is a migration guide from Flink <= 1.9, the option always was in Flink <= 1.9, so I think it is still worth mentioning it here.

[the remark this answers is not in the document]
Comment (KarmaGYZ) [reply docs/ops/memory/mem_migration.md]: Agreed. +1 to merge this PR.

**Comment by ptmagic** [comment]
Thank you for your help, I will continue to contribute

**Pull request closed without merging**